velox/exec/RowContainer.h

```diff
@@ -718,6 +744,15 @@ class RowContainer {
     return keyTypes_;
   }
 
+  inline void updateColumnMayHaveNulls(int32_t columnIndex, bool mayHaveNulls) {
```

Comment:
This is a public API? If so, please, document it.

My reading of the code suggests that by default columnsMayHaveNulls_ flags are set to 'false' and the callers are expected to call updateColumnMayHaveNulls to mark columns as maybe having nulls. If callers forget to do that, we'll end up with wrong results. If that's the case, it is very important to clearly document the responsibility of the callers to call this API before storing any values. I wonder why cannot we update these flags when storing values? Would that be more reliable?

Let's also update 'store' APIs to clarify that the caller is expected to call updateColumnMayHaveNulls with proper value before calling 'store'.

Reply:
> This is a public API? If so, please, document it.

Added the document, thanks.

> I wonder why cannot we update these flags when storing values? Would that be more reliable?

The reason I avoided doing this is that I thought it would lead to many duplicate updates and degrade performance. However, from my tests, I didn't observe any noticeable impact. Yes, it's better to update this information when modifying null bits. Updated, thanks.